src/components/input/Input.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { FC, forwardRef, ForwardedRef } from "react";
+import { StyledInput, Label } from "./Input.styled";
+
+export interface Props {
```

Comment:
One small suggestion to make this more flexible, we can reduce the interface to just `width`, `label`, `labelPlacement` and pass so change it to be:
```typescript
interface Props = {
   width?: string;
   label?: string;
   labelPlacement?: LabelPlacement;
 } & React.ComponentPropsWithoutRef<'input'>

```

and then pass them as `...delegated` like David suggested. What do you think?

Reply:
Awesome, love it.